File: 2338-minimum-consecutive-cards-to-pick-up/solution.py
class Solution:
    def minimumCardPickup(self, cards: List[int]) -> int:
        cards_last_seen = {}
        ans = math.inf
        for i in range(len(cards)):
            if cards[i] in cards_last_seen:
                ans = min(ans, i - cards_last_seen[cards[i]] + 1)
            cards_last_seen[cards[i]] = i
        return -1 if ans == math.inf else ans

File: 2338-minimum-consecutive-cards-to-pick-up/solution.java
class Solution {
    public int minimumCardPickup(int[] cards) {
        Map<Integer, Integer> last_seen = new HashMap<>();
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i < cards.length; ++i) {
            if (last_seen.containsKey(cards[i])) {
                ans = Math.min(ans, i - last_seen.get(cards[i]) + 1);
            }
            last_seen.put(cards[i], i);
        }
        return ans == Integer.MAX_VALUE ? -1: ans;
    }
}
